pkg/controller/ovsnodeosp/ovsnodeosp_controller.go

```diff
@@ -0,0 +1,323 @@
+			found.Spec = ds.Spec
+			err = r.client.Update(context.TODO(), found)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			r.setDaemonsetHash(instance, dsHash)
+			return reconcile.Result{RequeueAfter: time.Second * 10}, err
+		}
+	}
+
+	// Daemonset already exists - don't requeue
+	reqLogger.Info("Skip reconcile: Daemonset already exists", "Ds.Namespace", found.Namespace, "Ds.Name", found.Name)
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileOVSNodeOsp) setDaemonsetHash(instance *neutronv1.OVSNodeOsp, hashStr string) error {
+	if hashStr != instance.Status.DaemonsetHash {
+		instance.Status.DaemonsetHash = hashStr
+		if err := r.client.Status().Update(context.TODO(), instance); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func newDaemonset(cr *neutronv1.OVSNodeOsp, cmName string, templatesConfigHash string) *appsv1.DaemonSet {
+	var trueVar = true
+
+	daemonSet := appsv1.DaemonSet{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "DaemonSet",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cmName,
+			Namespace: cr.Namespace,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"daemonset": cr.Name + "-daemonset"},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"daemonset": cr.Name + "-daemonset"},
+				},
+				Spec: corev1.PodSpec{
+					NodeSelector:       common.GetComputeWorkerNodeSelector(cr.Spec.RoleName),
+					HostNetwork:        true,
+					HostPID:            true,
+					DNSPolicy:          "ClusterFirstWithHostNet",
+					Containers:         []corev1.Container{},
+					Tolerations:        []corev1.Toleration{},
+					ServiceAccountName: cr.Spec.ServiceAccount,
+					PriorityClassName:  "system-node-critical",
+				},
+			},
+		},
+	}
+
+	// add compute worker nodes tolerations
+	for _, toleration := range common.GetComputeWorkerTolerations(cr.Spec.RoleName) {
+		daemonSet.Spec.Template.Spec.Tolerations = append(daemonSet.Spec.Template.Spec.Tolerations, toleration)
+	}
+
+	containerSpec := corev1.Container{
+		Name:  "ovs-node-osp",
+		Image: cr.Spec.OvsNodeOspImage,
+		Command: []string{
+			"bash", "-c", "/usr/local/sbin/ovsnode.sh",
+		},
+		SecurityContext: &corev1.SecurityContext{
+			Privileged: &trueVar,
+		},
+		ReadinessProbe: &corev1.Probe{
+			Handler: corev1.Handler{
+				Exec: &corev1.ExecAction{
+					Command: []string{
+						"/usr/share/openvswitch/scripts/ovs-ctl", "status",
+					},
+				},
+			},
+			InitialDelaySeconds: 15,
+			PeriodSeconds:       5,
+		},
+		LivenessProbe: &corev1.Probe{
+			Handler: corev1.Handler{
+				Exec: &corev1.ExecAction{
+					Command: []string{
+						"/usr/share/openvswitch/scripts/ovs-ctl", "status",
+					},
+				},
+			},
+			InitialDelaySeconds: 15,
+			PeriodSeconds:       5,
+		},
+		Env: []corev1.EnvVar{
+			{
+				Name:  "TEMPLATES_CONFIG_HASH",
+				Value: templatesConfigHash,
+			},
```

Comment:
should we also add the SCRIPTS_CONFIG_HASH here to trigger restart when the scripts get updated, or maybe we don't want this behavior for the ovs-node-osp?